modules/store/src/selector.ts

```diff
@@ -17,13 +17,21 @@ export type ComparatorFn = (a: any, b: any) => boolean;
 
 export type DefaultProjectorFn<T> = (...args: any[]) => T;
 
+export type ProjectorFn<Args extends unknown[], T> = (...args: Args) => T;
+
+export type StrictMemoizedSelector<
```

Comment:
this type is not exported from the index so that developers do not create a dependency on this type (presuming it can one day be removed, and all memoized selectors have strict projectors).

instead, if a dev needs to declare the type of a variable, they can use the form `MemoizedSelector<State, Result, (slices...Slices) => Result>`